personal_budget_main.py

```diff
@@ -0,0 +1,214 @@
+        headers = ["Transaction_Date", "Transaction_Details", "Debit", "Credit"]
+    else:
+        headers = ["Month", "Transaction_Date", "Transaction_Details", "Debit", "Credit"]
+        if view_count is None:
+            view_count = len(report)
+        else:
+            cnt = 0
+            for i in range(len(report)):
+                if report[i][0]:
+                    cnt += 1
+                if cnt > view_count:
+                    view_count = i
+                    break
+
+    print(tabulate(report[:view_count], headers=headers, tablefmt="grid", numalign="right", floatfmt=".2f"))
+
+
+if __name__ == "__main__":
+    dir_path = os.path.split(os.path.realpath(__file__))[0] + "/source_data" # Path of directory where raw data is stored.
+    report_meta, sort_code, view_count = {}, {}, None
+
+    # Short options
+    short_options = "ht:s:o:c:"
+
+    # Long options
+    long_options = ["help", "report_type=", "sort_by=", "order_by=", "view_count="]
+
+    try:
+        # Parsing argument
+        arguments, values = getopt.getopt(sys.argv[1:], short_options, long_options)
+        
+        for currentArgument, currentValue in arguments:
+
+            if currentArgument in ("-h", "--help"):
+                print ("Displaying Help")
+                
+            elif currentArgument in ("-t", "--report_type"):
+                # Selecting output view based on user input.
+                temp = currentValue.lower().split("_")
+                if "month" in temp[0]:
+                    if "top" in temp:
+                        report_meta["type"] = "MT"
+                        report_meta["txn_per_month"] = int(temp[-1]) if temp[-1].isnumeric() else 3
+                        sort_code["sort_req"] = True
+                    else:
+                        report_meta["type"] = "M"
+                elif "transaction" in temp[0]:
+                    report_meta["type"] = "T"
+                    view_count = 20 if view_count is None else view_count
+                else:
+                    report_meta["type"] = "M"
+                
+            elif currentArgument in ("-s", "--sort_by"):
+                # Sort by column "D" : Debit, "C" : Credit, "S" : Surplus
+                sort_code["sort_req"] = True
+                sort_code["sort_by"] = "D" if "debit" in currentValue.lower() else "C" if "credit" in currentValue.lower() else "S"
+
+            elif currentArgument in ("-o", "--order_by"):
+                # Sort order Descending or Ascending
+                sort_code["order"] = "D" if any(sub in currentValue.lower() for sub in ["descend", "decreas"]) else "A"
+
+            elif currentArgument in ("-c", "--view_count"):
+                view_count = int(currentValue) if currentValue.isnumeric() else 10
+    
+    except getopt.error as error:
+        # Output error and return with an error code.
+        print(str(error))
+    
+    transactions_all, monthly_summary = extract_data_from_source(dir_path)
```

Comment:
transactions_all is not a Pythonic name. `transactions` should be enough or `all_transactions`